modules/miscutil/lib/errorlib.py

```diff
@@ -406,12 +416,25 @@ def register_exception(stream='error',
                                     'query_string': req.args,
                                     'headers': dict(req.headers_in),
                                     'https': req.is_https(),
-                                    'env': req.environ})
+                                    # Encode thanks to our custom and decode to let data intact for
+                                    # the invenio serializer.
+                                    #
+                                    # Indeed req.environ can containts C-Object which are non
+                                    # serializable.
+                                    'env': json.loads(json.dumps(req.environ, cls=SentryEncoder))
+                                     })
                 client.extra_context(user_info)
             filename = _get_filename_and_line(sys.exc_info())[0]
-            client.tags_context({'filename': filename},
-                                {'version': CFG_VERSION})
+            client.tags_context({'filename': filename, 'version': CFG_VERSION})
             client.captureException()
+        except Exception:
+            # Exception management of exception management
+            try:
+                client.captureException()
+            except Exception as err:
+                print >> sys.stderr, "Error in registering exception to '%s': '%s'" % (
```

Comment:
In next it will not fly: (1) it's not Python 3 friendly, (2) we use `current_app.logger.exception` instead custom `stderr` logging.